ggml.c

```diff
@@ -14755,7 +14755,7 @@ static void ggml_compute_backward(struct ggml_context * ctx, struct ggml_tensor
                     const int n_past = ((int32_t *) tensor->op_params)[0];
                     src0->grad =
                         ggml_add_or_set(ctx, src0->grad,
-                            ggml_diag_mask_zero_impl(ctx, tensor->grad, n_past, false),
+                            ggml_diag_mask_inf_impl(ctx, tensor->grad, n_past, false),
```

Comment:
I think `ggml_diag_mask_zero_impl` is the correct choice for backward pass of `GGML_OP_DIAG_MASK_INF`.
Since `GGML_OP_DIAG_MASK_INF` will set values to a fix value (infinite), the gradients for these values will be zero.
Using `ggml_diag_mask_inf_impl`, as proposed here, would make the gradients of these values also infinite, which will probably result in NaNs during training.

`GGML_OP_DIAG_MASK_INF` purpose is to mask out certain values, so that they don't influence the rest of the computations - likewise their gradients need to masked out, by zero-ing them.